packages/workbox-core/_private/quota.mjs

```diff
@@ -0,0 +1,74 @@
+/*
+ Copyright 2018 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import {logger} from './logger.mjs';
+import {assert} from './assert.mjs';
+
+import '../_version.mjs';
+
+const callbacks = new Set();
+
+/**
+ * Adds a function to the set of callbacks that will be executed when there's
+ * a quota error.
+ *
+ * @param {Function} callback
+ *
+ * @memberof workbox.core
+ */
+function registerCallback(callback) {
+  if (process.env.NODE_ENV !== 'production') {
+    assert.isType(callback, 'function', {
+      moduleName: 'workbox-core',
+      funcName: 'register',
+      paramName: 'callback',
+    });
+  }
+
+  callbacks.add(callback);
+
+  if (process.env.NODE_ENV !== 'production') {
+    logger.log('Registered a callback to respond to quota errors.', callback);
```

Comment:
How you checked to see how these function are logged (IIRC normal functions output their `toString()` result)? Maybe it makes more sense to output `.name` value?

Reply:
Yeah, it logs the code for the actual callback function. This will normally be

```js
() => this.deleteCacheAndMetadata()
```

or it might be the code for a custom callback, if you write and register your own.

I experimented with this a little bit and could go with just leaving out the logging of the function entirely, but I figured that more context for the development environment logging is better.

Since a developer could theoretically pass in whatever function they want here (in addition to using our automatically registered functions), I don't think we could count on `.name` being meaningful.

Reply:
I'd just like to leave it as-is to make sure we're providing the full context to folks who are in debug mode and reading the Console log messages.

Reply:
SG